libdnf/advisory/advisory_module.cpp

```diff
@@ -28,20 +28,11 @@ namespace libdnf::advisory {
 // AdvisoryModule
 AdvisoryModule::AdvisoryModule(AdvisoryModule::Impl * private_module) : p_impl(private_module) {}
 
-AdvisoryModule::AdvisoryModule(const AdvisoryModule & src) : p_impl(new Impl(*src.p_impl)) {}
+AdvisoryModule::AdvisoryModule(const AdvisoryModule & src) = default;
```

Comment:
You don't even need to declare these `= default`, you can remove them completely and they'll be auto-generated, won't they?

Reply:
That wouldn't work. `AdvisoryModule::Impl` is an incomplete type in the header file. So the methods must be implemented somewhere where the type is complete. -> in .cpp

I have an idea how to simplify it using a macro. So I can prepare a second solution. And then we will have a choice of what will be nicer.